btcec/schnorr/musig2/nonces.go

```diff
@@ -270,6 +270,16 @@ func GenNonces(options ...NonceGenOption) (*Nonces, error) {
 		return nil, err
 	}
 
+	// If the options contain a secret key, we XOR it with with the tagged
+	// random bytes.
+	if len(opts.secretKey) != 0 {
+		taggedHash := chainhash.TaggedHash(NonceAuxTag, randBytes[:])
+
+		for i, _ := range taggedHash {
```

Comment:
To increase readability this could just be a classic for loop up to 32 (can be `chainhash.HashSize` to not use a magic number).